pkg/control/client.go

```diff
@@ -134,11 +156,18 @@ func NewClient(ctx context.Context, cfg ClientConfig) (*Client, error) {
 	}
 
 	var (
-		gcc *grpc.ClientConn
-		err error
+		gcc  *grpc.ClientConn
+		edge pb.EdgeServicesClient
+		err  error
 	)
 
 	gClient := cfg.Client
+
+	if gClient == nil && cfg.GRPCConn != nil {
```

Comment:
If neither are passed in, then the client ends up in the 'unknownMode' mode. Previously it would end up just misconfigured, we could gate NewClient to make sure it has some connection to the server.

Reply:
How about a comment like
```
// We'll check for GRPCConn and Addr, and use that to set the gClient if they are available
// If neither are passed in, then the client ends up in the 'unknownMode' mode.
// [some short description of what unknown mode does?]